build-logic/src/main/kotlin/config-shade.gradle.kts

```diff
@@ -0,0 +1,41 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+import gradle.kotlin.dsl.accessors._d73e0aad27c892ff134862f94ec3182c.build
+import gradle.kotlin.dsl.accessors._d73e0aad27c892ff134862f94ec3182c.implementation
+import gradle.kotlin.dsl.accessors._d73e0aad27c892ff134862f94ec3182c.jar
+import org.gradle.kotlin.dsl.creating
+
+plugins {
+    id("com.github.johnrengelman.shadow")
```

Comment:
I mean, since there's already code duplication, shadow can still be used for bundling the shaded jar, the runtime relocation code can be kept. As for artifacts, they should depend on the regular jar publication and get the normal libraries in their classpath (even though I don't see the issue in Weave consumers using directly the relocated path, less work for you)